src/Microsoft.SqlTools.ServiceLayer/Connection/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -699,6 +701,8 @@ public class ServerInfo
             public string OsVersion;
 
             public string MachineName;
+
+            public Dictionary<string, object> Options { get; set; }
```

Comment:
I would recommend deriving from `GeneralRequestDetails` which contains an Options member a several nice related helper methods.  Then implementing getter\setters that read\write to the property bag.  This makes the internal code nice using strongly typed properties, but keeps the wire-format generic.  Take a look at `ConnectionDetails` for this pattern.